3. Import the required packages with their customary aliases.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

4. Use the %matplotlib inline magic command so that your pltos show in the notebook without having to call plt.show() every time.

In [2]:
%matplotlib inline

5. Using the pandas read_csv() method, read the GDP dataset into your notebook as a DataFrame called gdp_df. Take a look at the first 6 rows.

In [3]:
gdp_df = pd.read_csv("../data/gdp_percapita.csv")

In [4]:
gdp_df.head(6)

Country or Area  Year        Value  Value Footnotes
0     Afghanistan  2019  2065.036398              NaN
1     Afghanistan  2018  2033.779002              NaN
2     Afghanistan  2017  2058.383832              NaN
3     Afghanistan  2016  2057.062164              NaN
4     Afghanistan  2015  2068.265904              NaN
5     Afghanistan  2014  2102.385234              NaN

Answer 5: The first six rows of data are presented above. The data includes 4 columns - Country or Area, Year, Value, and Value Footnotes. Upon further inspection in Visual Studio Code, the data set concludes on row 6869 before skipping a row and listing footnotes. As a revision, the dataframe will be amended to include a provision with nrows to ensure that the footnote details are not picked up as part of the dataset. 

In [5]:
gdp_df = pd.read_csv("../data/gdp_percapita.csv", nrows=6869)

6. Repeat for the internet use dataset. Call this DataFrame internet_df. Take a look at the first six rows.

In [6]:
internet_df = pd.read_csv("../data/internet_use.csv")

ParserError: Error tokenizing data. C error: Expected 4 fields in line 4675, saw 6


Given the ParserError located in line 4675 I checked the rows in Visual Studio Code and determined that the data ends in line 4496 before footnotes begin in 4498. I will rerun the code to create the dataframe using nrows in order to truncate the dataset to exclude the footnote details.

In [7]:
internet_df = pd.read_csv("../data/internet_use.csv", nrows=4496)

In [8]:
internet_df.head(6)

Country or Area  Year     Value  Value Footnotes
0     Afghanistan  2014  6.390000              NaN
1     Afghanistan  2013  5.900000              NaN
2     Afghanistan  2012  5.454545              NaN
3     Afghanistan  2011  5.000000              NaN
4     Afghanistan  2010  4.000000              NaN
5     Afghanistan  2009  3.550000              NaN

Answer 6. The information in the internet use DataFrame has the same basic format as the gdp DataFrame.

7. Look at the shape of each DataFrame - how many rows? How many columns?

In [10]:
gdp_df.shape

(6869, 4)

In [11]:
internet_df.shape

(4496, 4)

Answer 7. Both dataframes are set up with 4 columns, as shown in the first 6 rows shown in Out[4] and Out[8]. However, the gdp data frame is significantly larger with 6869 rows compared to the 4496 rows in the internet data frame.

8. Take a look at the datatypes for the columns in each DataFrame.

In [13]:
gdp_df.dtypes

Country or Area     object
Year                object
Value              float64
Value Footnotes    float64
dtype: object

In [14]:
internet_df.dtypes

Country or Area     object
Year                object
Value              float64
Value Footnotes    float64
dtype: object

Answer 8. In both dataframes, Country or Area and Year are object data types and Value and Value Footnotes are float64 data types.

9. Take a look at the last 10 rows of each DataFrame in turn.

In [15]:
gdp_df.tail(10)

Country or Area      Year        Value  Value Footnotes
6859        Zimbabwe      1998  3653.920016              NaN
6860        Zimbabwe      1997  3580.048793              NaN
6861        Zimbabwe      1996  3520.430146              NaN
6862        Zimbabwe      1995  3226.413930              NaN
6863        Zimbabwe      1994  3263.934978              NaN
6864        Zimbabwe      1993  3033.504852              NaN
6865        Zimbabwe      1992  3054.889178              NaN
6866        Zimbabwe      1991  3426.598094              NaN
6867        Zimbabwe      1990  3324.348171              NaN
6868   footnoteSeqID  Footnote          NaN              NaN

In [16]:
internet_df.tail(10)

Country or Area      Year     Value  Value Footnotes
4486        Zimbabwe      2001  0.799846              NaN
4487        Zimbabwe      2000  0.401434              NaN
4488        Zimbabwe      1999  0.161676              NaN
4489        Zimbabwe      1998  0.081648              NaN
4490        Zimbabwe      1997  0.033080              NaN
4491        Zimbabwe      1996  0.016790              NaN
4492        Zimbabwe      1995  0.007684              NaN
4493        Zimbabwe      1994  0.001739              NaN
4494        Zimbabwe      1990  0.000000              NaN
4495   footnoteSeqID  Footnote       NaN              NaN

Answer 9. The last 10 rows for each dataframe can be observed in Out[15] and Out[16]. Both final rows are related to footnotes rather than the data. This is due to a calculation error on my part as python starts counting at row 0 rather than row 1. This can be amended by recalculating the dataframe with a different value for nrows.